pkg/controller/storagecluster/update.go

```diff
@@ -104,6 +127,69 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string)
 	return c.syncNodes(cluster, oldPodsToDelete, []string{}, hash)
 }
 
+// function that returns if a pod belongs to a node that is running kvdb
+func isPodInKvdbNode(pod *v1.Pod, nodes []string, nodeToStoragePods map[string][]*v1.Pod) bool {
```

Comment:
Every pod has a node name in its spec - `pod.Spec.NodeName`. We can just use that and compare it with the kvdb nodes. Rather `getKVDBNodeAvailability()` can just return a map for kvdb nodes for easy look up.

The above code could be simplified to - 
```
_, isKvdbNode := kvdbNodes[pod.Spec.NodeName]
if cluster.Spec.Kvdb != nil && cluster.Spec.Kvdb.Internal && isKvdbNode {
...
}
```